### PR TITLE
Make main-tf contents exported

### PR DESCRIPTION
**Pull request by therealmitchconnors**
This change will allow users of upjet to test their examples against known good terraform files, instead of having to apply them e2e, which can involve creating expensive resources.  @ytsarev @stevendborrelli

### CODE REVIEW

**Comment by jeanduplessis** [comment]
@therealmitchconnors thank you for the contribution. Would you mind signing the CLA please: https://cla.upbound.io/upbound/upjet?pullRequest=223

**Comment by therealmitchconnors** [comment]
@jeanduplessis is there a reason upjet doesn't use the CNCF cla as part of the crossplane project?  Each additional cla requires separate review from my legal team, which is a big ask for a small pr like this...

**Comment by jeanduplessis** [comment]
@therealmitchconnors Unfortunately, due to Upjet not being part of the /crossplane/ org but the /upbound/ org in GitHub, it cannot use the same CLA.

**Comment by therealmitchconnors** [comment]
Now that upbound is a CNCF project, will my CLA work?

**Comment by jeanduplessis** [comment]
@therealmitchconnors we'll be moving this repo to the /crossplane/ GitHub org soon which will no longer require a CLA, and only require users to sign off their commits for [DCO compliance](https://wiki.linuxfoundation.org/dco). 

If you are able to update your commits on this PR with a sign off I can accept them without need for CLA sign off.

**Comment by therealmitchconnors** [comment]
signoff complete

File: pkg/terraform/files.go [hunk 1]
```diff
@@ -126,6 +124,12 @@ func (fp *FileProducer) WriteMainTF() (ProviderHandle, error) {
 			},
 		},
 	}
+}
+
+// WriteMainTF writes the content main configuration file that has the desired
+// state configuration for Terraform.
+func (fp *FileProducer) WriteMainTF() error {
```
Comment (ulucinar) [inline pkg/terraform/files.go:131]:
```suggestion
func (fp *FileProducer) WriteMainTF() (ProviderHandle, error) {
```

File: pkg/terraform/files.go [hunk 2]
```diff
@@ -91,9 +91,7 @@ type FileProducer struct {
 	fs          afero.Afero
 }
 
-// WriteMainTF writes the content main configuration file that has the desired
-// state configuration for Terraform.
-func (fp *FileProducer) WriteMainTF() (ProviderHandle, error) {
+func (fp *FileProducer) BuildMainTF() map[string]any {
```
Comment (ulucinar) [inline pkg/terraform/files.go:94]: Let's add some documentation for this exported function. Thank you.

**Review by ulucinar — COMMENTED** [review]
Thank you @therealmitchconnors for the PR. Could you please also rebase it?

**Pull request merged**